deepinv/training/trainer.py

```diff
@@ -518,6 +536,14 @@ def setup_train(self, train=True, **kwargs):
             )
 
         _ = self.load_model()
+        # Cache whether model.forward accepts 'update_parameters' to avoid per-call inspect
```

Comment:
the risk with a global flag is that it persists after training. what if the user loads a model later that isn't updated?